tokeniser.go

```diff
@@ -17,38 +18,60 @@ const (
 
 	// double represents "double quoted" parts of a BAPS3 message.
 	double
+
+	// bufsize is the number of bytes the Tokeniser will try to read from
+	// its Reader in one go.
+	bufsize int = 4096
```

Comment:
I'm not sure I like this. Apart from 4096 being ridiculously large, it just doesn't seem like an efficient way of doing it.
I feel like reading a single byte at a time is better, as per http://grokbase.com/t/gg/golang-nuts/139fgmycba/go-nuts-reading-a-single-byte-from-io-reader-into-a-variable#20130915mlerpyqkq3gvbvnm2ifryvu2cm - I don't think 75ns per byte is something to be concerned about (it's likely to be limited by the network connection anyway)